fido2/attestation.py

```diff
@@ -182,27 +185,54 @@ def verify(self, statement, auth_data, client_data_hash):
 
 
 OID_AAGUID = x509.ObjectIdentifier("1.3.6.1.4.1.45724.1.1.4")
+OID_AIK_CERTIFICATE = x509.ObjectIdentifier("2.23.133.8.3")
 
 
-def _validate_attestation_certificate(cert, aaguid):
+def _validate_attestation_certificate(
+    cert,
+    aaguid,
+    empty_subject=False,
+    has_subject_alternative_name=False,
+    has_aik_certificate=False,
+):
     if cert.version != x509.Version.v3:
         raise InvalidData("Attestation certificate must use version 3!")
-    c = cert.subject.get_attributes_for_oid(x509.NameOID.COUNTRY_NAME)
-    if not c:
-        raise InvalidData("Subject must have C set!")
-    o = cert.subject.get_attributes_for_oid(x509.NameOID.ORGANIZATION_NAME)
-    if not o:
-        raise InvalidData("Subject must have O set!")
-    ous = cert.subject.get_attributes_for_oid(x509.NameOID.ORGANIZATIONAL_UNIT_NAME)
-    if not ous:
-        raise InvalidData('Subject must have OU = "Authenticator Attestation"!')
-
-    ou = ous[0]
-    if ou.value != "Authenticator Attestation":
-        raise InvalidData('Subject must have OU = "Authenticator Attestation"!')
-    cn = cert.subject.get_attributes_for_oid(x509.NameOID.COMMON_NAME)
-    if not cn:
-        raise InvalidData("Subject must have CN set!")
+    if not empty_subject:
```

Comment:
could probably worth spliting that in two instructions "check_subject" and "does_not_have_subject"